src/Illuminate/Support/Arr.php

```diff
@@ -62,19 +63,24 @@ public static function build($array, callable $callback)
     /**
      * Collapse an array of arrays into a single array.
      *
-     * @param  array  $array
+     * @param  \Traversable|array  $array
      * @return array
      */
     public static function collapse($array)
     {
+        // Iterate on the array for performance.
```

Comment:
Actually, `Collection` return its internal iterator of `$this->items`, so `foreach` on Collection is loop on $items